v2/datastream-to-spanner/terraform/samples/mysql-sharded-end-to-end/README.md

```diff
@@ -296,6 +303,40 @@ can exclude it from the state file using `terraform state rm` command.
 
 ## FAQ
 
+### Running end to end migration using live and bulk templates combined
+
+The recommendation for running a production end-to-end migration
+is using a combination of live and bulk Terraform templates.
+This optimizes for speed and cost-effectiveness, especially when dealing
+with large historical datasets. The live Terraform template can handle the
+entire migration process, including historical data. However, migrating
+large historical datasets solely with the live template can be slow and
+expensive.
+
+The steps to follow are:
+
+1. **Change Data Capture Infra setup**
+    - Begin by running the live Terraform template with the variables
+        - `skip_dataflow` = true (This disables the creation of the Dataflow job for CDC data)
+        - `enable_backfill` = false (This disables the backfill functionality within datastream)
+    - This step sets up the necessary infrastructure for the migration but skips the dataflow job creation.
+
+2. **Bulk Migration**
+    - Run the [Bulk Terraform template](../../../../sourcedb-to-spanner/terraform/samples/README.md). This template will
+      create Dataflow job(s) specifically designed for migrating
+      historical data.
+    - Monitor the Dataflow job until it completes. This step migrates the bulk of your historical data efficiently.
+
+3. **Live Migration (with Dataflow)**
+    - Once the bulk migration is complete, run the live Terraform template again, this time with the following
+      variables:
+        - `skip_dataflow` = false (This enables the creation of the Dataflow job for CDC data)
+        - `enable_backfill` = false (Keep backfill disabled as it was handled in the bulk migration)
+    - This step creates a Dataflow job that migrates the CDC data captured by Datastream.
+
+Once Spanner is up to speed with the source instance, turn off the source instance and wait for Spanner to fully
+catch up. Once caught up, cut over the application to use Spanner.
+
```

Comment:
I wonder if we should have this as a dedicated README in `spanner-common` instead of as an FAQ in `datastream-to-spanner` template - seems too important. WDYT? It covers multiple templates, so it makes sense. 

We should have a diagram/flow chart to describe the overall process and leave room for enhancing it in the future via additional sections (completing the migration, validation etc). Looks to me that it deserves its known README. In future, it can be converted to Cloud docs.

In these FAQ, you can maybe explain the specific params and refer to the other README?

Reply:
Agreed, this will require a much more detailed guide, likely in cloud docs. I can move it to spanner-common but will someone ever open a spanner-common readme by themselves? Maybe we should have 2 duplicates, both in live and bulk. For the non-wanderers, we will anyways point to proper documentation.

Reply:
> Agreed, this will require a much more detailed guide, likely in cloud docs.

Agreed, this should be a first-class citizen in cloud docs.

> I can move it to spanner-common but will someone ever open a spanner-common readme by themselves? Maybe we should have 2 duplicates under a major heading in the readme itself, both in live and bulk. For the non-wanderers, we will anyways point to proper documentation.

My opinion on this lies in the first part itself, I do not think customers will self-discover this on their own until this documentation is part of the cloud docs. With that said, I would recommend placing this in a guide of its own (which I think we agree on). Between duplicating it between bulk and live v/s placing in spanner-common, I think the added discoverability is incremental compared to cloud docs, so I would recommend placing in spanner-common, but I am okay with duplication too, as long as it is it's own dedicated `README` file.

We should also add the diagram/flow pieces as well.